Assets/MixedRealityToolkit.Extensions/Socketer/Scripts/SocketEndpoint.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+
+namespace Microsoft.MixedReality.Toolkit.Extensions.Experimental.Socketer
+{
+    public sealed class SocketEndpoint
+    {
+        private enum ConnectionState
+        {
+            Connected,
+            Disconnected
+        }
+
+        /// <summary>
+        /// IP address for the socket endpoint
+        /// </summary>
+        public string Address { get; private set; }
+
+        private readonly SocketerClient socketerClient;
+        private readonly int sourceId;
+        private ConcurrentQueue<IncomingMessage> incomingQueue;
+        private DateTime lastActiveTimestamp;
+        private TimeSpan timeoutInterval;
+
+        private ConnectionState State { get; set; } = ConnectionState.Connected;
+
+        /// <summary>
+        /// Returns true if the socket endpoint is connected, otherwise false
+        /// </summary>
+        public bool IsConnected
+        {
+            get { return State == ConnectionState.Connected; }
+        }
+
+        /// <summary>
+        /// Call to set the socket endpoint state to disconnected
+        /// </summary>
+        public void Disconnect()
+        {
+            State = ConnectionState.Disconnected;
```

Comment:
i dont actually see a diconnection occuring here.

Reply:
will add disconnect logic and update the example